### DISCUSSION TITLE
SMILES Parse Error: syntax error while parsing

### DISCUSSION THREAD

**Discussion by danielmanu93** [rdkit/rdkit#4246] [Development]
Hi I'm having this issue. I'm trying to generate molecules from smiles with the attached generate function but I get the "SMILES parse error" in the terminal when I call the function. How can I fix this issue?

![1](https://user-images.githubusercontent.com/67971334/121933026-c42f6080-cd02-11eb-897c-8151d6661602.jpeg)

![2](https://user-images.githubusercontent.com/67971334/121933033-c691ba80-cd02-11eb-8da8-dd3f7d826291.jpeg)

![3](https://user-images.githubusercontent.com/67971334/121947281-1d9f8b80-cd13-11eb-8e36-ece0cc92a140.jpeg)

**Comment by bp-kelley** [comment]
You are not reading a smiles file, this looks like a python pickle format, please see:

https://docs.python.org/3/library/pickle.html

for more details, but I would expect something like:

```
import pickle
mols = pickle.load(open(filename, 'rb'))
```

might do the trick.

**Reply by danielmanu93** [reply]
Thanks, I have figured it out from your reply.
